File: 3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.cpp
class Solution {
public:
    int numberOfSpecialChars(string s) {
        int res = 0;
        unordered_set<char> found;
        for(int i = 0; i < s.length(); i++)
        {
            if(!found.count(s[i]))
            {
                if(s[i] >= 'a' && s[i] <= 'z')
                {
                    if(found.count(s[i] - 32))
                    {
                        res++;
                    }
                    found.insert(s[i]);
                }
                else
                {
                    if(found.count(s[i] + 32))
                    {
                        res++;
                    }
                    found.insert(s[i]);
                }
            }
        }
        return res;
    }
};

File: 3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.py
class Solution:
    def numberOfSpecialChars(self, word: str) -> int:
        res = 0
        found = set()
        for ch in word:
            if ch not in found:
                if ch.islower():
                    if ch.upper() in found:
                        res += 1
                else:
                    if ch.lower() in found:
                        res += 1
                found.add(ch)
        return res